lib/matplotlib/backends/backend_nbagg.py

```diff
@@ -242,6 +242,12 @@ def trigger_manager_draw(manager):
 
     @staticmethod
     def show(*args, **kwargs):
+        if args or {*kwargs} - {"block"}:
```

Comment:
Couldn't that be simply
~~~
def show(*args, *, block=None, **kwargs):
    if args or kwargs:
~~~

Note also that block is not used within the method.

Reply:
But it's legal to do `show(block=...)`; sure, `block` is not used but I don't want to emit a deprecation in that case as that makes it harder to write backend-independent code (or if you really want to warn, it should be something like "the block parameter has no effect in the nbagg backend").

Reply:
Not sure what official best practices are here, but I would make all `show()` methods identical in their API. They override `_Backend.show()` and should formally be substitutable. Any non-working parameters should be warned about and not be ignored or rejected with a type error.

Reply:
Oh sorry, I missed that you proposed changing the signature.  Yes, that works, pushed.